BEIMA.Backend.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,227 @@
+            //For cloud testing, this will throw an exception, and it will use the env variables
+            //defined in beima.yml.
+            try
+            {
+                string[] paths = { "..", "..", "..", "..", "BEIMA.Backend", "local.settings.json" };
+                string combinedPath = Path.Combine(paths);
+                string fullPath = Path.GetFullPath(combinedPath);
+                var settings = JsonConvert.DeserializeObject<LocalSettings>(
+                    File.ReadAllText(fullPath));
+
+                foreach (var setting in settings.Values)
+                {
+                    Environment.SetEnvironmentVariable(setting.Key, setting.Value);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+            }
+            
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+
+        [Test]
+        public void InitialState_Action_ExpectedResult()
+        {
+            Assert.IsTrue(true);
+        }
+
+        [Test]
+        public void ConnectorNotCreated_CallConstructorFiveTimes_FiveInstancesAreEqual()
+        {
+            //Tests to make sure singleton is working
+            var connector1 = MongoConnector.Instance;
+            var connector2 = MongoConnector.Instance;
+            var connector3 = MongoConnector.Instance;
+            var connector4 = MongoConnector.Instance;
+            var connector5 = MongoConnector.Instance;
+            Assert.IsTrue(connector1 == connector2);
+            Assert.IsTrue(connector1 == connector3);
+            Assert.IsTrue(connector1 == connector4);
+            Assert.IsTrue(connector1 == connector5);
+        }
+
+        [Test]
+        public void ConnectorNotCreated_CallConstructorOnce_InstanceConnectedToCorrectEnvironment()
+        {
+            var mongo = MongoConnector.Instance;
+            if (Environment.GetEnvironmentVariable("CurrentEnv") == "dev-local")
+            {
+                Assert.IsTrue(mongo.CurrentServerType == ServerType.Local);
+            }
+            else
+            {
+                Assert.IsTrue(mongo.CurrentServerType == ServerType.Cloud);
+            }
+        }
+
+        [Test]
+        public void ConnectorCreated_GetDeviceGivenValidDeviceId_DeviceDocumentReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //Setup (inserting a document)
+            BsonDocument doc = new BsonDocument {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.IsTrue(insertResult is ObjectId);
```

Comment:
```suggestion
            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
```
Again, just me being picky, but I generally like to avoid using Assert.IsTrue if possible.

Reply:
Fixed [d10ddbb](https://github.com/cs481-ekh/s22-beima/pull/67/commits/d10ddbbb704bdd673bd1c99f2a7b83230349ca7b)